include/dynd/types/tuple_type.hpp

```diff
@@ -145,13 +147,13 @@ namespace ndt {
     }
 
     /** Makes a tuple type with the specified types */
-    static type make(const nd::array &field_types, bool variadic = false)
+    static type make(std::vector<type> field_types, bool variadic = false)
     {
       return type(new tuple_type(field_types, variadic), false);
```

Comment:
Is it worth changing the 'make' signature to take an rvalue reference?

/** Makes a tuple type with the specified types */
static type make(std::vector<type> &&field_types, bool variadic = false)
{
  return type(new tuple_type(std::move(field_types), variadic), false);
}

The drawback is that this would require quite a bit of code changes, like https://github.com/libdynd/libdynd/blob/master/src/dynd/types/datashape_parser.cpp#L953 ::

  ndt::type pos = ndt::tuple_type::make(std::move(pos_arg_types));
  ndt::type kwds = ndt::struct_type::make(kw_names, kw_arg_types);
  ndt::type result_tp = ndt::tuple_type::make(std::move({pos, kwds}));